### PR TITLE
Configure eslint for mocha globals

### PR DESCRIPTION
**Pull request by bergarces**
- Added `.eslintrc.json` files to stop commit hooks from wrongly flagging errors.
- Fixed some of the flagged errors in test files.
- Not all errors from all packages have been fixed. Truffle test files that make use of global variables such as `artifacts` are still affected.
- An alternative to adding a `.eslintrc.json` file to every root folder, would be to add the filepath to the mocha exception list in the root `.eslintrc.json`.

### CODE REVIEW

File: packages/box/.eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
+  "overrides": [
+    {
+      "files": ["./test/**/*.js"],
+      "env": {
+        "mocha": true
+      }
+    }
+  ]
+}
```
Comment (gnidan) [inline packages/box/.eslintrc.json:10]: New line at end of file :)

File: packages/compile-solidity/.eslintrc.json [hunk 2]
```diff
@@ -0,0 +1,10 @@
+{
+  "overrides": [
+    {
+      "files": ["./test/**/*.js"],
+      "env": {
+        "mocha": true
+      }
+    }
+  ]
+}
```
Comment (gnidan) [inline packages/compile-solidity/.eslintrc.json:10]: New line at end of file :)

File: packages/compile-vyper/.eslintrc.json [hunk 3]
```diff
@@ -0,0 +1,10 @@
+{
+  "overrides": [
+    {
+      "files": ["./test/**/*.js"],
+      "env": {
+        "mocha": true
+      }
+    }
+  ]
+}
```
Comment (gnidan) [inline packages/compile-vyper/.eslintrc.json:10]: New line at end of file (don't break my `cat` please!)

File: packages/contract-schema/.eslintrc.json [hunk 4]
```diff
@@ -0,0 +1,10 @@
+{
+  "overrides": [
+    {
+      "files": ["./test/**/*.js"],
+      "env": {
+        "mocha": true
+      }
+    }
+  ]
+}
```
Comment (gnidan) [inline packages/contract-schema/.eslintrc.json:10]: new line at end of file 🦜

File: packages/contract-tests/.eslintrc.json [hunk 5]
```diff
@@ -0,0 +1,10 @@
+{
+  "overrides": [
+    {
+      "files": ["./test/**/*.js"],
+      "env": {
+        "mocha": true
+      }
+    }
+  ]
+}
```
Comment (gnidan) [inline packages/contract-tests/.eslintrc.json:10]: y'know maybe just search and replace them all :)

**Review by gnidan — CHANGES_REQUESTED** [review]
This looks great, apart from the comment you'll see a lot of :)

Thanks for doing this! It'll be nice to have working editor integration

**Comment by bergarces** [comment]
> This looks great, apart from the comment you'll see a lot of :)
> 
> Thanks for doing this! It'll be nice to have working editor integration

Added newlines and changed my editor settings.

**Review by gnidan — APPROVED** [review]
Looks good!

**Pull request merged**